warehouse/integrations/postgres/load.go

```diff
@@ -61,7 +67,7 @@ type LoadUsersTable struct {
 	LoadFileDownloader downloader.Downloader
 }
 
-func (lt *LoadTable) Load(ctx context.Context, tableName string, tableSchemaInUpload model.TableSchema) (string, error) {
+func (lt *LoadTable) Load(ctx context.Context, tableName string, tableSchemaInUpload model.TableSchema) (LoadTableResponse, error) {
```

Comment:
It is a preferred pattern to begin the transaction externally and pass it as an argument to this method.
Having one method creating a transaction whose lifecycle needs to be managed by the caller can be considered to be a weak, error-prone design.
Typically we prefer having clear transaction boundaries, e.g. `WithTx(func(tx *sql.Tx) error {})`, so that transaction can be consistently commited/rolledback as it is fit in all scenarios.

Reply:
Good point. I was also thinking of passing transactions but because of the complexity of the code, I took a short route.
Also, I have started a [discussion](https://rudderlabs.slack.com/archives/C01HTT66UMB/p1682341076776989) around the decoupling of identifies and users table loading so that transactions won't be required.

let me see how we can pass transactions.